sdk/test/trace/sampler_benchmark.cc

```diff
@@ -0,0 +1,128 @@
+#include "opentelemetry/sdk/trace/samplers/always_off.h"
+#include "opentelemetry/sdk/trace/samplers/always_on.h"
+#include "opentelemetry/sdk/trace/samplers/parent_or_else.h"
+#include "opentelemetry/sdk/trace/samplers/probability.h"
+
+#include <cstdint>
+
+#include <benchmark/benchmark.h>
+
+namespace
+{
+using opentelemetry::sdk::trace::AlwaysOffSampler;
+using opentelemetry::sdk::trace::AlwaysOnSampler;
+using opentelemetry::sdk::trace::ParentOrElseSampler;
+using opentelemetry::sdk::trace::ProbabilitySampler;
+using opentelemetry::sdk::trace::Decision;
+using opentelemetry::trace::SpanContext;
+
+void BM_AlwaysOffSamplerConstruction(benchmark::State &state)
```

Comment:
Does it make sense to bench the constructors? Especially for always on/off?  They only have default constructors.

Reply:
When I spoke to Oulin she mentioned that having a base noop sampler to compare the more advanced samplers to was a good idea. I figured these would work well for this purpose.

Reply:
Makes sense.  Perhaps leave a comment?

Reply:
Sounds good. Added